front/src/modules/SimulationReportSheet/utils/formatSimulationReportSheet.ts

```diff
@@ -337,3 +349,82 @@ export const getStopType = (stopType: StdcmStopTypes | undefined, t: TFunction<'
   !stopType
     ? t('reportSheet.serviceStop')
     : capitalizeFirstLetter(t(`trainPath.stopType.${stopType}`));
+
+export const transformStepsToApiFormat = (
+  steps: StdcmPathStep[],
+  beforeTolerance: string,
+  afterTolerance: string
+): Step[] =>
+  steps.map((step) => {
+    const baseStep: Step = {
+      duration: 0,
+      location: {
+        primary_code: step.location?.uic ? formatUicToCi(step.location.uic) : '',
+        secondary_code: step.location?.secondary_code || '',
+      },
+    };
+
+    if (step.isVia) {
+      const stopTypeMap: Record<StdcmStopTypes, StepType> = {
+        [StdcmStopTypes.PASSAGE_TIME]: 'P',
+        [StdcmStopTypes.DRIVER_SWITCH]: 'R',
+        [StdcmStopTypes.SERVICE_STOP]: 'S',
+        [StdcmStopTypes.OVERTAKE]: 'P',
+      };
+
+      return {
+        ...baseStep,
+        duration: step.stopFor?.ms || 0,
+        type: stopTypeMap[step.stopType] || 'P',
+      };
+    } else {
```

Comment:
You don't need this else since you returned just above
```suggestion
    }
```

Reply:
Done